realm-server/tests/indexing-test.ts

```diff
@@ -140,12 +162,105 @@ module("indexing", function (hooks) {
       {
         instancesIndexed: 1,
         definitionsBuilt: 0,
-        modulesAnalyzed: 0,
         instanceErrors: 0,
         definitionErrors: 0,
+        moduleErrors: 0,
+      },
+      "indexed correct number of files"
+    );
+  });
+
+  test("can recover from a card error after error is removed from card source", async function (assert) {
+    // introduce errors into 2 cards and observe that invalidation doesn't
+    // blindly invalidate all cards are in an error state
+    await realm.write(
+      "pet.gts",
+      `
+          import { contains, field, Card } from "https://cardstack.com/base/card-api";
+          import StringCard from "https://cardstack.com/base/string";
+          export class Pet extends Card {
+            @field firstName = contains(StringCard);
+          }
+          throw new Error('boom!');
+        `
+    );
+    assert.deepEqual(
+      realm.searchIndex.stats,
+      {
+        instancesIndexed: 0,
+        definitionsBuilt: 0,
+        instanceErrors: 1,
+        definitionErrors: 1,
+        moduleErrors: 1,
+      },
+      "indexed correct number of files"
+    );
+    await realm.write(
+      "person.gts",
+      `
+          // syntax error
+          export class IntentionallyThrownError {
+        `
+    );
+    assert.deepEqual(
+      realm.searchIndex.stats,
+      {
+        instancesIndexed: 0,
+        definitionsBuilt: 0,
+        instanceErrors: 3, // 1 post, 2 persons
+        definitionErrors: 2, // post, fancy person (person is not a def error because the syntax failed)
```

Comment:
if you visit a file and there is a module, lets write out an error doc regardless